packages/fondue/src/components/TextInput/TextInput.tsx

```diff
@@ -125,246 +126,261 @@ const ExtraActionButton: ForwardRefRenderFunction<HTMLButtonElement | null, Extr
 };
 const ExtraActionButtonWithRef = forwardRef(ExtraActionButton);
 
-export const TextInput = ({
-    id: propId,
-    type = TextInputType.Text,
-    decorator,
-    validation = Validation.Default,
-    clearable = false,
-    placeholder,
-    required,
-    obfuscated,
-    disabled = false,
-    autocomplete = false,
-    dotted = false,
-    value = '',
-    copyable = false,
-    onChange,
-    onEnterPressed,
-    onBlur,
-    onClear,
-    onFocus,
-    size,
-    spellcheck,
-    readonly,
-    focusOnMount,
-    selectable = false,
-    extraActions,
-    maxLength,
-    ...props
-}: TextInputProps): ReactElement => {
-    const { isFocusVisible: inputIsFocusVisible, focusProps: inputFocusProps } = useFocusRing({ isTextInput: true });
-    const { isFocusVisible: clearButtonIsFocusVisible, focusProps: clearButtonFocusProps } = useFocusRing();
-    const { isFocusVisible: passwordButtonIsFocusVisible, focusProps: passwordButtonFocusProps } = useFocusRing();
-    const { isFocusVisible: copyButtonIsFocusVisible, focusProps: copyButtonFocusProps } = useFocusRing();
+export const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
+    (
+        {
+            id: propId,
+            type = TextInputType.Text,
+            decorator,
+            validation = Validation.Default,
+            clearable = false,
+            placeholder,
+            required,
+            obfuscated,
+            disabled = false,
+            autocomplete = false,
+            dotted = false,
+            value = '',
+            copyable = false,
+            onChange,
+            onEnterPressed,
+            onBlur,
+            onClear,
+            onFocus,
+            size,
+            spellcheck,
+            readonly,
+            focusOnMount,
+            selectable = false,
+            extraActions,
+            maxLength,
+            ...props
+        },
+        ref,
+    ): ReactElement => {
+        const inputElement = useRef<HTMLInputElement | null>(null);
+        useImperativeHandle(ref, () => inputElement.current!, []);
```

Comment:
Is the imperative handle needed? We are returning the entire ref and not a subset of it.

I think we can remove both the handle and the `inputElement` internal ref, and just proxy the `ref` coming to the component directly to the DOM node?

Reply:
We have useEffect which sets the focus on the element thats why is Imperative handle there. Do you suggest to change it?

Reply:
okay, my bad, the GitHub diff showed that part as a change of this PR, but I can see now that it was already there, sorry! 🙏🏻